MarkupsToModel/Logic/vtkSlicerMarkupsToModelLogic.cxx

```diff
@@ -538,6 +577,373 @@ void vtkSlicerMarkupsToModelLogic::UpdateOutputHermiteSplineModel(vtkMRMLMarkups
+{
+  vtkPoints* points = markupsPointsPolyData->GetPoints();
+  if (points == NULL)
+  {
+    vtkErrorMacro("Markups contains a vtkPoints object that is null.");
+    return;
+  }
+
+  int numPoints = points->GetNumberOfPoints();
+  // redundant error checking, to be safe
+  if (numPoints < LINE_MIN_NUMBER_POINTS)
+  {
+    vtkErrorMacro("Not enough points to compute polynomial parameters. Need at least " << LINE_MIN_NUMBER_POINTS << " points but " << numPoints << " are provided.");
+    return;
+  }
+
+  if (markupsPointsParameters->GetNumberOfTuples())
+  {
+    // this should never happen, but in case it does, output a warning
+    vtkWarningMacro("markupsPointsParameters already has contents. Clearing.");
+    while (markupsPointsParameters->GetNumberOfTuples()) // clear contents just in case
+      markupsPointsParameters->RemoveLastTuple();
+  }
+
+  for (int v = 0; v < numPoints; v++)
+  {
+    markupsPointsParameters->InsertNextTuple1( v / double(numPoints-1) );
+    // division to clamp all values to range 0.0 - 1.0
+  }
+}
+
+//------------------------------------------------------------------------------
+void vtkSlicerMarkupsToModelLogic::ComputePointParametersMinimumSpanningTree(vtkPolyData * markupsPointsPolyData, vtkDoubleArray* markupsPointsParameters)
+{
+  vtkPoints* points = markupsPointsPolyData->GetPoints();
+  if (points == NULL)
+  {
+    vtkErrorMacro("Markups contains a vtkPoints object that is null.");
+    return;
+  }
+
+  int numPoints = points->GetNumberOfPoints();
+  // redundant error checking, to be safe
+  if (numPoints < LINE_MIN_NUMBER_POINTS)
+  {
+    vtkErrorMacro("Not enough points to compute polynomial parameters. Need at least " << LINE_MIN_NUMBER_POINTS << " points but " << numPoints << " are provided.");
+    return;
+  }
+
+  // vtk boost algorithms cannot be used because they are not built with 3D Slicer
+  // so this is a custom implementation of:
+  // 1. constructing an undirected graph as a 2D array
+  // 2. Finding the two vertices that are the farthest apart
+  // 3. running prim's algorithm on the graph
+  // 4. extract the "trunk" path from the last vertex to the first
+  // 5. based on the distance along that path, assign each vertex a polynomial parameter value
+
+  // in the following code, two tasks are done:
+  // 1. construct an undirected graph
+  std::vector<double> distances(numPoints*numPoints);
+  distances.assign(numPoints*numPoints, 0.0);
+  // 2. find the two farthest-seperated vertices in the distances array
+  int treeStartIndex = 0;
+  int treeEndIndex = 0;
+  double maximumDistance = 0;
+  // iterate through all points
+  for (int v = 0; v < numPoints; v++)
+  {
+    for (int u = 0; u < numPoints; u++)
+    {
+      double pointU[3], pointV[3];
+      points->GetPoint(u, pointU);
+      points->GetPoint(v, pointV);
+      double distX = (pointU[0] - pointV[0]); double distXsq = distX * distX;
```

Comment:
don't put multiple statements in one line, it's just not nice style (makes the code difficult to read)